araproc/analysis/standard_reco.py

```diff
@@ -924,3 +926,346 @@ def get_avg_corr_snr(self, wave_packet, excluded_channels = []):
+
+                else: 
+                    delay = xcorr_times[ 
+                        np.nanargmax(xcorr_volts[zoomed_indices]) # index of max xcorr in zoomed array
+                        + zoomed_indices[0] # Adjusted by first zoomed_index
+                    ]
+
+            # AraRoot always compares channels with smaller IDs to channels with 
+            #   larger IDs but we always want to compare to the reference channel.
+            # Correct for this if the current ch_ID is larger than the reference ch_ID
+            if ch_ID > reference_ch: 
+                delay *= -1
+            
+            # Save the calculated arrival delay
+            delays[ch_ID] = delay
+
+        return delays
+
+    def __trim_array(self, times, values, trim):
+        """
+        Trim an array by `trim` values, being careful to avoind trimming out the
+        maximum signal
+
+        Parameters
+        ----------
+        times : Iterable
+        values : Iterable
+        trim : int
+
+        Returns
+        -------
+        times : Iterable
+        values : Iterable
+        """
+
+        # Soft exit if the user doesn't actually request a trim
+        if trim == 0: 
+            return times, values
+        
+        # Identify the index where the peak signal is held
+        value_max_idx = np.nanargmax(values)
+
+        # Calculate amount to trim off beginning and end of the waveform, 
+        #   initially triming equal parts off the front and back unless the 
+        #   index with maximal signal is in one of those regions
+        front_trim = trim // 2
+        back_trim = int( np.ceil( trim / 2 ) )
+        if value_max_idx - front_trim < 0:
+            # The peak of the signal is within the front region of the waveform 
+            #   about to be trimmed. Switch to trimming off the back entirely.
+            front_trim = 0
+            back_trim = trim
+        elif back_trim - value_max_idx < 0:
+            # The peak of the signal is within the back region of the waveform
+            #   about to be trimmed. Switch to trimming off the front entirely.
+            front_trim = trim
+            back_trim = 0
+
+        # Return the results
+        if back_trim == 0:
+            return times[front_trim:], values[front_trim:]
+        else: 
+            return times[front_trim:-back_trim], values[front_trim:-back_trim]
+
+    def get_csw(
+        self, data, useful_event, solution, polarization, reco_results,
+        excluded_channels=[], which_distance='distant', return_csw=False
+    ):
+        """
+        Build the Coherently Summed Waveform (CSW) for the given `useful_event`
+          by shifting each waveform accoriding to the time delay calculated 
+          after finding the max cross correlation between two channels 
+          in a window around the expected arrival delay. 
+
+        Parameters
+        ----------
+        data : AnalysisDataset
+        useful_event : UsefulAtriStationEvent
```

Comment:
Reminder to remove `useful_event` from arguments and just pass whether its a soft trigger.

Reply:
Yep thanks!